### ISSUE TITLE
`tailscale cert` fails when using k8s Secret storage due to lack of statedir

### ISSUE THREAD

**Issue by KevinM2k** [tailscale/tailscale#3974]
### What is the issue?

In k8s pod when running the command:

```
tailscale cert <domain>
``` 

expect a 200, instead get the response:

```
# tailscale --socket=/tmp/tailscaled.sock cert <domain>
HTTP 500 Internal Server Error: failed to get cert dir
 (expected 200)
```

Looking in to the code, it looks like the cert dir should be where the state file is... however as its k8s the state file is being saved inside a secret so the cert dir isnt actually set. i'm no go expert so could be wrong here.

### Steps to reproduce

- Use the method in the docs/k8s folder and use the `make proxy` command to create a pod that dnats to the correct service.
- Enable https inside tailscale
- Then run the tailscale cert command on this pod

### OS

Linux

### Tailscale version

latest

**Comment by lorenzo95** [comment]
Ran into the same issue. with latest and decided to try using unstable instead. Getting a different error message there.

`cert # tailscale version
1.21.82
  tailscale commit: 8267ea0f8094ddc325877e3db91b3b06756e7312
  go version: go1.17.6-ts25fe91a25c
/cert # tailscale --socket=/tmp/tailscaled.sock cert domain
unexpected output: no delimiter
`

**Comment by DentonGentry** [comment]
Duplicate of https://github.com/tailscale/tailscale/issues/5676 which was fixed by https://github.com/tailscale/tailscale/pull/7142

The fix is in current 1.37.x unstable builds, and will be in the 1.38 release.
